presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java

```diff
@@ -96,7 +99,7 @@ public CacheStatsMBean getCursorCacheStats()
         return new CacheStatsMBean(cursorProcessors);
     }
 
-    public Supplier<CursorProcessor> compileCursorProcessor(RowExpression filter, List<RowExpression> projections, Object uniqueKey)
+    public Supplier<CursorProcessor> compileCursorProcessor(Optional<RowExpression> filter, List<? extends RowExpression> projections, Object uniqueKey)
```

Comment:
Technically speaking changing type for `projections` arg should not be part of this commit.
I will not cry if you leave it here though.